File: 2022.01.21/python/1018.py
n,m = map(int,input().split())
board = [list(input()) for _ in range(n)]
result = 10**18*9
for i in range(n-7):
    for j in range(m-7):
        idx1 = 0 # 맨 왼쪽 위가 블랙이라 가정
        idx2 = 0 # 맨 왼쪽 위가 화이트라 가정
        for r in range(i,i+8):
            for c in range(j,j+8):
                if (r+c)%2 == 0:
                    if board[r][c] == "W" : idx1 += 1
                    else: idx2 += 1
                else:
                    if board[r][c] == "B" : idx1 += 1
                    else: idx2+=1
        result = min(result,idx1,idx2)
print(result)

File: 2022.01.21/javascript/1018.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');
const [n,m] = input[0].split(' ').map(Number);
const board = input.slice(1,input.length-1);
const checkBoardChange = (pattern,cnt,r,c) => {
    for (let i = 0 ; i<8 ; i++){
        for (let j = 0 ; j<8 ; j++){
            if ((i+j)%2 == 0 && board[r+i][c+j] != pattern) cnt++;
            if ((i+j)%2 != 0 && board[r+i][c+j] == pattern) cnt++;
        }
    }
    return cnt
}
let result = 10**18*9, one, two;
const patterns = new Map();
patterns.set("W","B");
patterns.set("B","W");
for (let i = 0 ; i<n-7 ; i++){
    for (let j = 0 ; j<m-7 ; j++){
        one = checkBoardChange(board[i][j],0,i,j);
        two = checkBoardChange(patterns.get(board[i][j]),0,i,j);
        result = Math.min(one,two,result);
    }
}
console.log(result)